BUG: 3 tests fail at home, not in GHA

These tests:

```py
FAILED hkl/tests/test_save_restore_UB.py::test_restore_orientation - KeyError: "Wrong axes names.  Expected ['john', 'paul', 'george', 'ringo'], received ('omega', 'chi', 'phi', 'tth')"
FAILED hkl/tests/test_user.py::test_calc_UB - TypeError: Expected list, tuple, or calc.Position() object, received FourcRealPos(omega=-145.451, chi=90, phi=0, tth=69.0966)
FAILED hkl/tests/test_user.py::test_setor - TypeError: Expected list, tuple, or calc.Position() object, received FourcRealPos(omega=-145.451, chi=90, phi=0, tth=69.0966)
```

--- a/hkl/sample.py
+++ b/hkl/sample.py
@@ -370,7 +370,7 @@ class HklSample(object):
                 if pos is None:
                     # so use the current motor positions
                     return False
-                elif type(pos).__name__.startswith("Pos"):
+                elif type(pos).__name__.startswith("Pos") or type(pos).__name__.endswith("RealPos"):
                     # This is (probably) a calc.Position namedtuple
                     if False in [isinstance(v, (int, float)) for v in pos]:
                         raise TypeError(f"All values must be numeric, received {pos!r}")

--- a/hkl/util.py
+++ b/hkl/util.py
@@ -444,19 +444,23 @@ def restore_reflections(orientation, diffractometer):
         Diffractometer object.
     """
     _check_geometry(orientation, diffractometer)
+    calc = diffractometer.calc
     # remember this wavelength
-    wavelength0 = diffractometer.calc.wavelength
+    wavelength0 = calc.wavelength
 
     # short aliases
     pseudos = orientation["_pseudos"]
     reals = orientation["_reals"]
+    if reals != calc.physical_axis_names and reals == calc._geometry.axis_names_get():
+        # Substitute user-defined axes names for canonical axes names.
+        reals = calc.physical_axis_names
     orientation_reflections = []
     # might be renamed axes
-    renaming = diffractometer.calc._axis_name_to_original
+    renaming = calc._axis_name_to_original
 
     for ref_base in orientation["reflections_details"]:
         # every reflection has its own wavelength
-        diffractometer.calc.wavelength = ref_base["wavelength"]
+        calc.wavelength = ref_base["wavelength"]
 
         # Order of the items is important.
         # Can't just use the dictionaries in ``orientation``.
@@ -471,17 +475,17 @@ def restore_reflections(orientation, diffractometer):
 
         # assemble the final form of the reflection for add_reflection()
         reflection = tuple([*miller_indices, ppp])
-        r = diffractometer.calc.sample.add_reflection(*reflection)
+        r = calc.sample.add_reflection(*reflection)
         if ref_base["orientation_reflection"]:
             orientation_reflections.append(r)
 
     if len(orientation_reflections) > 1:
         # compute **UB** from the last two orientation reflections
-        diffractometer.calc.sample.compute_UB(*orientation_reflections[-2:])
+        calc.sample.compute_UB(*orientation_reflections[-2:])
 
     # restore previous wavelength
-    if diffractometer.calc.wavelength != wavelength0:
-        diffractometer.calc.wavelength = wavelength0
+    if calc.wavelength != wavelength0:
+        calc.wavelength = wavelength0
 
 
 def restore_orientation(orientation, diffractometer):

